crates/ide-diagnostics/src/lib.rs

```diff
@@ -139,13 +141,36 @@ impl Default for ExprFillDefaultMode {
     }
 }
 
-#[derive(Default, Debug, Clone)]
+#[derive(Debug, Clone)]
 pub struct DiagnosticsConfig {
     pub proc_macros_enabled: bool,
     pub proc_attr_macros_enabled: bool,
     pub disable_experimental: bool,
     pub disabled: FxHashSet<String>,
     pub expr_fill_default: ExprFillDefaultMode,
+    // FIXME: We may want to include a whole `AssistConfig` here
+    pub insert_use: InsertUseConfig,
+}
+
+impl Default for DiagnosticsConfig {
+    fn default() -> Self {
+        Self {
+            proc_macros_enabled: Default::default(),
+            proc_attr_macros_enabled: Default::default(),
+            disable_experimental: Default::default(),
+            disabled: Default::default(),
+            expr_fill_default: Default::default(),
+            // FIXME: We should not implement default for DiagnosticConfig, but many tests
+            // need it currently. Maybe replace default impl with a `sample_config_for_tests`.
+            insert_use: InsertUseConfig {
+                granularity: ImportGranularity::Preserve,
+                enforce_granularity: false,
+                prefix_kind: PrefixKind::Plain,
+                group: false,
+                skip_glob_imports: false,
+            },
+        }
+    }
```

Comment:
Let's replace this with an associated constant that is cfg(test) then, that's what we do for other test configs I believe

Reply:
`disabled` field is a hashset, which can't be initialized on const context currently. Is a function with `#[cfg(test)]` ok, or you would prefer lazy_static/once_cell here?

Reply:
A function is good then in that case, a static doesn't benefit us here